### PR TITLE
XWIKI-15495: Enable CTRL+CLICK on document tree to open links in new tabs

### PR DESCRIPTION
**Pull request by slauriere**
- When ctrl key is pressed, open link in new window / tab in tree.js  event listener

### CODE REVIEW

File: xwiki-platform-core/xwiki-platform-tree/xwiki-platform-tree-webjar/src/main/webjar/tree.js [hunk 1]
```diff
@@ -493,7 +493,11 @@ define(['jquery', 'JobRunner', 'jsTree', 'tree-finder'], function($, JobRunner)
       } else if (data.event && !$(data.event.target).hasClass('jstree-no-link') &&
           $(data.event.target).closest('.jstree-no-links').length === 0) {
         // The node selection was triggered by an user event and links are enabled.
-        window.location.href = selectedNode.a_attr.href;
+        if (data.event.ctrlKey === true) {
```
Comment (vmassol) [inline xwiki-platform-core/xwiki-platform-tree/xwiki-platform-tree-webjar/src/main/webjar/tree.js:496]: maybe a quick comment saying "When pressing CTRL and clicking on a document tree entry, open the entry on a new tab"?

Comment (slauriere) [reply xwiki-platform-core/xwiki-platform-tree/xwiki-platform-tree-webjar/src/main/webjar/tree.js:497]: Thanks Vincent, I added a comment in the latest commit indeed

**Comment by slauriere** [comment]
PS: I tested the change successfully by deploying the newly built jar locally

**Comment by vmassol** [comment]
Seems good to me but I'll let @mflorea provide feedback and merge it.

File: xwiki-platform-core/xwiki-platform-tree/xwiki-platform-tree-webjar/src/main/webjar/tree.js [hunk 2]
```diff
@@ -493,7 +493,12 @@ define(['jquery', 'JobRunner', 'jsTree', 'tree-finder'], function($, JobRunner)
       } else if (data.event && !$(data.event.target).hasClass('jstree-no-link') &&
           $(data.event.target).closest('.jstree-no-links').length === 0) {
         // The node selection was triggered by an user event and links are enabled.
-        window.location.href = selectedNode.a_attr.href;
+        // When pressing Ctrl key and clicking on a document tree entry that is a link, open link in new window / tab
```
Comment (mflorea) [inline xwiki-platform-core/xwiki-platform-tree/xwiki-platform-tree-webjar/src/main/webjar/tree.js:496]:
```suggestion
        // When pressing Ctrl key and clicking on a tree node that has a link, open the link in new window / tab.
```
This code is for the **generic** tree. There's no notion of document at this level. The document tree macro uses this code later on.

**Review by mflorea — CHANGES_REQUESTED** [review]

File: xwiki-platform-core/xwiki-platform-tree/xwiki-platform-tree-webjar/src/main/webjar/tree.js (hunk 2 above)
Comment (slauriere) [reply xwiki-platform-core/xwiki-platform-tree/xwiki-platform-tree-webjar/src/main/webjar/tree.js]: Thank you Marius for the review, I updated this code comment accordingly.

**Review by mflorea — APPROVED** [review]

**Pull request merged**